bin/BuildPackages.sh

```diff
@@ -60,36 +30,51 @@ then
         "Type: cd ../pkg; ../bin/BuildPackages.sh"
 fi
 
-# CURDIR
 CURDIR="$(pwd)"
+GAPDIR="$(cd .. && pwd)"
+COLORS=yes
+PACKAGES=
+
+# If output does not go into a terminal (but rather into a log file),
+# turn of colors.
+[[ -t 1 ]] || COLORS=no
+
+while [[ "$#" -ge 1 ]]; do
+  option="$1" ; shift
+  case "$option" in
+    --with-gaproot)   GAPDIR="$1"; shift ;;
+    --with-gaproot=*) GAPDIR=${option#--with-gaproot=}; shift ;;
+    --no-color)       COLORS=no ;;
+    *)                PACKAGES="$PACKAGES $option" ;;
+  esac
+done
 
-# GAPDIR
-GAPDIR=""
-if [[ "$#" -ge 1 ]]
+# packages to build
+if [[ -z "$PACKAGES" ]]
 then
-  case "$1" in
-    --with-gaproot=*)
-      GAPDIR=$(sed 's|--with-gaproot=||' <<< "$1")
-      shift
-    ;;
-
-    *)
-      cd ..
-      GAPDIR="$(pwd)"
-      cd "$CURDIR"
-    ;;
-  esac
+  PACKAGES="$(find . -maxdepth 2 -type f -name PackageInfo.g  | xargs -n 1 dirname)"
```

Comment:
Why is it better to parse the output of ````find```` than to use 
````
shopt -s nullglob
PACKAGES=(*)
shopt -u nullglob
````
which should be the proper way to get all directories. Or you do not want to do the ````PackageInfo.g```` check later?